src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentsAnalyzer.php

```diff
@@ -268,7 +268,12 @@ public static function analyze(
 
             $inferred_arg_type = $statements_analyzer->node_data->getType($arg->value);
 
-            if (null !== $inferred_arg_type && null !== $template_result && null !== $param && null !== $param->type) {
+            if (null !== $inferred_arg_type
+                && null !== $template_result
+                && null !== $param
+                && null !== $param->type
+                && !$arg->unpack
```

Comment:
Was two failling tests here:
- ArrayFunctionCallTest::testValidCode(splatArrayIntersect)
- FunctionTemplateTest::testValidCode(splatTemplateParam)

And I've added `&& !$arg->unpack`.
I don't know how to fill template result when arg is being unpacking.
This branch needs for this kind inference: https://psalm.dev/r/086e0799b6
I think we can leave it as is.